qa/tasks/ceph_deploy.py

```diff
@@ -263,14 +263,28 @@ def ceph_volume_osd_create(ctx, config):
         no_of_osds = 0
         for remote in osds.remotes.iterkeys():
             # all devs should be lvm
-            devs = teuthology.get_scratch_devices(remote)
             osd_create_cmd = './ceph-deploy osd create ' + remote.shortname + ' '
             # default is bluestore so we just need config item for filestore
-            if config.get('filestore') is not None:
-                osd_create_cmd += '--filestore '
-                # filestore with ceph-volume also needs journal, #todo
+            roles = ctx.cluster.remotes[remote]
+            dev_needed = len([role for role in roles
+                              if role.startswith('osd')])
+            all_devs = teuthology.get_scratch_devices(remote)[0:dev_needed]
+            devs = all_devs[0:dev_needed]
+            # rest of the devices can be used for journal if required
+            jdevs = dev_needed
             for device in devs:
-                osd_create_cmd += ' --data ' + device
+                device_split = device.split('/')
+                lv_device = device_split[-1]
```

Comment:
is `device` here something like `/dev/vg/lv` ? because this would give you `lv` in that case

Reply:
sorry i was refreshing, but yes the device is from the list and is of format: /dev/vg_hdd/lv_4